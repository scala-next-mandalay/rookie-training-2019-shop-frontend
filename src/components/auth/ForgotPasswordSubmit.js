import React from 'react';
import PropTypes from 'prop-types';
import { Box, Container, Link, Button, TextField,Grid}  from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import '../../assets/style.css';
import { green } from '@material-ui/core/colors';
import { makeStyles,createMuiTheme } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { ThemeProvider } from '@material-ui/styles';
 import { FormattedMessage } from 'react-intl';



const useStyles = makeStyles(theme => ({
  btnWrapper: {
    marginTop: theme.spacing(2),
    position: 'relative',
  },
  btnProgress: {
    color: green[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12,
  },
}));

const ForgotPasswordSubmit = React.memo(({
  authState, 
  changeAuthState,
  loading,
  error,
  email,
  forgotPasswordSubmit,
  history
}) => {
  const classes = useStyles();
  const [form, setForm] = React.useState({ password:"",confirmationCode:""});
  
  const handleChangeValue = fieldName => event => {
    const newForm = {...form};
    newForm[fieldName] =  event.target.value;
    setForm(newForm);
  };
  
  const handleSubmit = event => {
    event.preventDefault();
    forgotPasswordSubmit(email, form['confirmationCode'], form['password']);
  };
  const handleClose=event=>{
    event.preventDefault();
    history.push("/");
  };
  
  const theme = createMuiTheme({
  palette: {
    primary: {
       main:'#cccccc'
    }
  },
});
  
  const content = (
  <Container component="main" maxWidth="xs" className="container">
  <Box className="authTitle">
  <FormattedMessage id="Label.CheckEmail" defualtMessage="Please check your email" />
  </Box>
  
  
  <Link onClick={handleClose}>
  <span className="close-btn">
  <img src="https://cdn4.iconfinder.com/data/icons/miu/22/circle_close_delete_-128.png" alt="closeicon"></img> 
   </span>
   </Link>

  

  <form onSubmit={handleSubmit}>
      <Box display="flex" mx="auto" my="auto" mt={2} fontWeight={600} color="error.main">
            {error}
      </Box>
      <Box mt={3}>
      <ThemeProvider theme={theme}>
        <Grid container>
        <Grid item xs={12} sm={12}  >
        <Box mx="auto" my="auto" p={1} mt={2}>
            <TextField
              id="confirmationCode"
              label="Confirmation Code"
              onChange={handleChangeValue("confirmationCode")} 
              value={form.confirmationCode}
              variant="outlined"
              required
              fullWidth
               InputLabelProps={{
              style: {
              color: "#ffffff",
              }
             }}
             
            />
          
             </Box>
        </Grid>
        <Grid item xs={12} sm={12}  >
         <Box mx="auto" my="auto"  p={1} mt={2}>
            <TextField
              id="password"
              label="New Password"
              type="password"
              autoComplete="new-password"
              onChange={handleChangeValue("password")}
              value={form.password}
              variant="outlined"
              required
              fullWidth
               InputLabelProps={{
              style: {
              color: "#ffffff",
              }
             }}
            />
            </Box>
        </Grid>
        </Grid>
        </ThemeProvider>
        </Box>
           <Grid container className={classes.btnWrapper}>
              <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              disabled={loading}
            >
            
              <FormattedMessage id="Button.Confirm" defualtMessage="Confirm" />
            </Button>
              {loading && <CircularProgress size={24} className={classes.btnProgress} />}
             </Grid>  
             <Grid container >
              <Grid item xs={12}>
              <Box mt={2}>
                 <Box color="red">
                  <FormattedMessage id="Label.ResendCode" defualtMessage="Resend code to" />{email}
                </Box>
                </Box>
              </Grid>
            </Grid>
     </form>
    </Container>
  );
  return (authState === 'forgotPasswordReset') ? content : null;
});

ForgotPasswordSubmit.propTypes = {
  authState: PropTypes.string.isRequired,
  changeAuthState: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  email: PropTypes.string,
  forgotPasswordSubmit: PropTypes.func.isRequired,
};

export default withRouter(ForgotPasswordSubmit);