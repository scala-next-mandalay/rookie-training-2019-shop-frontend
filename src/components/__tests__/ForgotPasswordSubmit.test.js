import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import ForgotPasswordSubmit from '../auth/ForgotPasswordSubmit';
import Parent from '../Parent';
/*global jest*/
/*global expect*/
/*global MouseEvent*/
/*global Event*/
let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});



const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
let authState = "forgotPasswordReset";
const changeAuthState = jest.fn((signIn) => {
  authState = signIn;
});

const loading = false;
const error = "Forgot password ";
let email = "a@gmail.com";
const forgotPasswordSubmit = jest.fn((email, code, password) => {
  let code1 = code;
});

describe("ForgotPasswordSubmit component snapshot", () => {
  it('matches the snapshot', () => {
    const ForgotPasswordSubmitSnapshot = renderer.create(
        <Parent>
      <ForgotPasswordSubmit
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        email={email}
        forgotPasswordSubmit={forgotPasswordSubmit}
      /></Parent>
    ).toJSON();
    expect(ForgotPasswordSubmitSnapshot).toMatchSnapshot();

  });

})

describe("Testing ForgotPasswordSubmit", () => {
  it('Testing textfied and  confirm button ', () => {
    act(() => {
      ReactDOM.render(
          <Parent>
        <ForgotPasswordSubmit
          authState={authState}
          changeAuthState={changeAuthState}
          loading={loading}
          error={error}
          email={email}
          forgotPasswordSubmit={forgotPasswordSubmit}
        /></Parent>, container);

    });

    const inputArr = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr[0], '1234');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });

    expect(inputArr[0].value).toBe('1234');
    act(() => {
      valueSetter.call(inputArr[1], 'abc11111');
      inputArr[1].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[1].value).toBe('abc11111');
    
    const buttonArr = document.querySelectorAll('button');
    act(() => {
      buttonArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(forgotPasswordSubmit).toHaveBeenCalledWith("a@gmail.com","1234","abc11111");
  })

 it('Testing textfied and  confirm button with no data ', () => {
    act(() => {
      ReactDOM.render(
          <Parent>
        <ForgotPasswordSubmit
          authState={authState}
          changeAuthState={changeAuthState}
          loading={loading}
          error={error}
          email={email}
          forgotPasswordSubmit={forgotPasswordSubmit}
        /></Parent>, container);

    });

    const inputArr = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr[0], '');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });

    expect(inputArr[0].value).toBe('');
    act(() => {
      valueSetter.call(inputArr[1], '');
      inputArr[1].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[1].value).toBe('');
    
    const buttonArr = document.querySelectorAll('button');
    act(() => {
      buttonArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(forgotPasswordSubmit).not.toHaveBeenCalledWith( );
  });
})