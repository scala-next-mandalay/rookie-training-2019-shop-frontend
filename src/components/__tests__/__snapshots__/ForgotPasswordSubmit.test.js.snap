// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordSubmit component snapshot matches the snapshot 1`] = `
<main
  className="MuiContainer-root container MuiContainer-maxWidthXs"
>
  <div
    className="MuiBox-root MuiBox-root-11 authTitle"
  >
    <span>
      Please check your email
    </span>
  </div>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <span
      className="close-btn"
    >
      <img
        alt="closeicon"
        src="https://cdn4.iconfinder.com/data/icons/miu/22/circle_close_delete_-128.png"
      />
    </span>
  </a>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-48"
    >
      Forgot password 
    </div>
    <div
      className="MuiBox-root MuiBox-root-49"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiBox-root MuiBox-root-154"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="confirmationCode"
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              >
                Confirmation Code
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="confirmationCode"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-212 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-213"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiBox-root MuiBox-root-214"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="password"
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              >
                New Password
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="new-password"
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-212 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-213"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-btnWrapper-1 MuiGrid-container"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          <span>
            Confirm
          </span>
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiBox-root MuiBox-root-338"
        >
          <div
            className="MuiBox-root MuiBox-root-339"
          >
            <span>
              Send Code to 
            </span>
            a@gmail.com
          </div>
        </div>
      </div>
    </div>
  </form>
</main>
`;
